cmd/sonobuoy/app/results.go

```diff
@@ -0,0 +1,240 @@
+				os.Exit(1)
+			}
+		},
+		Args: cobra.ExactArgs(1),
+	}
+	AddPluginFlag(&data.plugin, cmd.Flags())
+	cmd.Flags().StringVarP(
+		&data.mode, "mode", "m", resultModeReport,
+		`Modifies the format of the output. Valid options are report, detailed, or dump.`,
+	)
+	cmd.Flags().StringVarP(
+		&data.node, "node", "n", "",
+		`Traverse results starting at the node with the given name. Defaults to the real root.`,
+	)
+	cmd.Flags().BoolVarP(
+		&data.skipPrefix, "skip-prefix", "s", false,
+		`When printing items linking to files, only print the file contents.`,
+	)
+
+	return cmd
+}
+
+func getReader(filepath string) (*results.Reader, func(), error) {
+	f, err := os.Open(filepath)
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "could not open sonobuoy archive: %v", filepath)
+	}
+
+	gzr, err := gzip.NewReader(f)
+	if err != nil {
+		return nil, func() { f.Close() }, errors.Wrap(err, "could not make a gzip reader")
+	}
+
+	r := results.NewReaderWithVersion(gzr, results.VersionTen)
+	return r, func() { gzr.Close(); f.Close() }, nil
+}
+
+func result(input resultsInput) error {
+	r, cleanup, err := getReader(input.archive)
+	defer cleanup()
+	if err != nil {
+		return err
+	}
+
+	// If we want to dump the whole file, don't decode to an Item object first.
+	if input.mode == resultModeDump {
+		fReader, err := r.PluginResultsReader(input.plugin)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get results reader for plugin %v", input.plugin)
+		}
+		_, err = io.Copy(os.Stdout, fReader)
+		return err
+	}
+
+	// For summary and detailed views, get the item as an object to iterate over.
+	obj, err := r.PluginResultsItem(input.plugin)
+	if err != nil {
+		return err
+	}
+
+	obj = getItemInTree(obj, input.node)
+	if obj == nil {
+		return fmt.Errorf("node named %q not found", input.node)
+	}
+
+	switch input.mode {
+	case resultModeDetailed:
+		printResultsDetails([]string{}, obj, input)
+	default:
+		printResultsSummary(obj)
+	}
+
+	return nil
+}
+
+func getItemInTree(i *results.Item, root string) *results.Item {
+	if i == nil {
+		return nil
+	}
+
+	if root == "" || i.Name == root {
+		return i
+	}
+
+	if len(i.Items) > 0 {
+		for _, v := range i.Items {
+			subItem := getItemInTree(&v, root)
+			if subItem != nil {
+				return subItem
+			}
+		}
+	}
+
+	return nil
+}
+
+func printResultsDetails(treePath []string, o *results.Item, input resultsInput) {
+	if o == nil {
+		return
+	}
+
+	if len(o.Items) > 0 {
+		treePath = append(treePath, o.Name)
+		for _, v := range o.Items {
+			printResultsDetails(treePath, &v, input)
+		}
+		return
+	}
+
+	if o.File == "" {
+		// Print each leaf node as a json object. Add the path as a metadata field for access by the end user.
+		if o.Metadata == nil {
+			o.Metadata = map[string]string{}
+		}
+		o.Metadata["path"] = strings.Join(treePath, "|")
+		b, err := json.Marshal(o)
+		if err != nil {
+			logrus.Errorf("Error marshalling item to json: %v", err)
```

Comment:
By logging these errors rather than returning them and letting them bubble up, the error from `result` will be `nil` which will affect the overall exit code. Is this intended?

Reply:
meh, I dont think I spent that much time thinking about it. I'll go ahead and just bubble everything up. This only means that if one node is poorly formed then we abort instead of continuing but I dont think that is that big of a deal.